add-friends-likes

Novacancy531 opened the pull request:
Реализация ФЗ-11

Novacancy531 committed:
Разработка:
1. Изменена архитектура приложения.
2. Добавлена новая логика.
- добавление пользователей в друзья.
- Проставление/удаление лайков для фильмов.
3. Дополнено логирование.

Novacancy531 committed:
Изменения:
Конструкторы заменены на аннотацию в классах контроллерах и сервисах.

VadimZharkov commented on src/main/java/ru/yandex/practicum/filmorate/controller/GlobalExceptionHandler.java:
```diff
@@ -30,27 +33,31 @@ public ResponseEntity<ErrorResponse> handleValidationException(MethodArgumentNot
-        public record ErrorResponse(LocalDateTime timestamp, int status, String error, String message) {
+    public record ErrorResponse(LocalDateTime timestamp, int status, String error, String message) {
     }
 }
```
Не реализовано.
"Убедитесь, что ваше приложение возвращает корректные HTTP-коды:
...   
404 — для всех ситуаций, если искомый объект не найден;
500 — если возникло исключение."

VadimZharkov commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,42 @@
+import java.util.HashMap;
+
+@Component
+@Data
```
`@Data` даст доступ через геттеры-сеттеры к внутренним полям класса всеь желающим, для реализации требуемой функциональности такой видимости не требуется.

VadimZharkov commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.HashMap;
+
+@Component
+@Data
```
Тоже самое, `@Data` лишнее

VadimZharkov reviewed (changes requested):
Нужно исправить замечания.

Novacancy531 committed:
Разработка:
1. Добавлено исключение NotFoundException.java.
2. В классах хранилищах убрана аннотация @Data.
3. Дополнена обработка ошибок в классах сервисах.

Novacancy531 committed:
Исправления:
Изменено для соответствия CheckStyle.
